### PR TITLE
[SPARK-35045][SQL] Add an internal option to control input buffer in univocity

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR makes the input buffer configurable (as an internal option). This is mainly to work around uniVocity/univocity-parsers#449.

### Why are the changes needed?

To work around uniVocity/univocity-parsers#449.

### Does this PR introduce _any_ user-facing change?

No, it's only internal option.

### How was this patch tested?

Manually tested by modifying the unittest added in https://github.com/apache/spark/pull/31858 as below:

```diff
diff --git a/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala b/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala
index fd25a79619d..b58f0bd3661 100644
--- a/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala
+++ b/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala
@@ -2460,6 +2460,7 @@ abstract class CSVSuite
       Seq(line).toDF.write.text(path.getAbsolutePath)
       assert(spark.read.format("csv")
         .option("delimiter", "|")
+        .option("inputBufferSize", "128")
         .option("ignoreTrailingWhiteSpace", "true").load(path.getAbsolutePath).count() == 1)
     }
   }
```

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala [hunk 1]
```diff
@@ -211,6 +211,8 @@ class CSVOptions(
   }
   val lineSeparatorInWrite: Option[String] = lineSeparator
 
+  val inputBuffer: Option[Int] = parameters.get("inputBuffer").map(_.toInt)
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala:214]: maybe `inputBufferSize`?

**Review by MaxGekk — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala (hunk 1 above)
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala:214]:
```suggestion
  val inputBufferSize: Option[Int] = parameters.get("inputBuffer").map(_.toInt)
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala [hunk 2]
```diff
@@ -257,6 +259,7 @@ class CSVOptions(
     settings.setIgnoreLeadingWhitespaces(ignoreLeadingWhiteSpaceInRead)
     settings.setIgnoreTrailingWhitespaces(ignoreTrailingWhiteSpaceInRead)
     settings.setReadInputOnSeparateThread(false)
+    inputBuffer.foreach(settings.setInputBufferSize)
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala:262]:
```suggestion
    inputBufferSize.foreach(settings.setInputBufferSize)
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala [hunk 3]
```diff
@@ -211,6 +211,8 @@ class CSVOptions(
   }
   val lineSeparatorInWrite: Option[String] = lineSeparator
 
+  val inputBufferSize: Option[Int] = parameters.get("inputBuffer").map(_.toInt)
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala:214]:
```suggestion
  val inputBufferSize: Option[Int] = parameters.get("inputBufferSize").map(_.toInt)
```

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.1/3.0.
Thank you @HyukjinKwon .

**Comment by HyukjinKwon** [comment]
Thx Max.

**Pull request closed without merging**